src/Network/MQTT/Parser.hs

```diff
@@ -160,7 +160,33 @@ parseSubscribe :: Word8 -> Parser SubscribePacket
 parseSubscribe = undefined
 
 parseSuback :: Word8 -> Parser SubackPacket
-parseSuback = undefined
+parseSuback flags = do
+  assert (flags == 0x00) "Reserved bits MUST be set to zero"
+
+  len <- fromIntegral <$> remainingLength
+  assert (len > 2) "Remaining length is too short"
+
+  packetId <- packetIdentifier
+  responses <- count (len - 2) packetResponse
+
+  return SubackPacket
+    { subackPacketIdentifier = packetId
+    , subackResponses        = responses
+    }
+
+  where
+    packetResponse :: Parser (Maybe QoS)
+    packetResponse = do
+      response <- anyWord8
+      assert (response `elem` [ 0x00, 0x01, 0x02, 0x80 ])
```

Comment:
I would use `response <= 0x02 || response == 0x80` as that's faster than iterating over the list.